zebra-consensus/src/transaction.rs

```diff
@@ -575,13 +600,13 @@ where
                     miner_fee,
                     legacy_sigop_count,
                 },
-                Request::Mempool { transaction, .. } => {
+                Request::Mempool { transaction: ref tx, .. } => {
                     let transaction = VerifiedUnminedTx::new(
-                        transaction,
-                        miner_fee.expect(
-                            "unexpected mempool coinbase transaction: should have already rejected",
-                        ),
+                        tx.clone(),
+                        miner_fee.expect("fee should have been checked earlier"),
                         legacy_sigop_count,
+                        #[cfg(any(test, feature = "proptest-impl"))]
+                        req.skip_checks().is_some_and(|checks| checks.contains(&SkipCheck::Zip317))
```

Comment:
I am not sure why but i remember seeing other odd behaviours when using `proptest-impl`. Here is one optionb to fix it:


```suggestion
                        #[cfg(any(test, feature = "proptest-impl"))]
                        req.skip_checks().is_some_and(|checks| checks.contains(&SkipCheck::Zip317)),
                        #[cfg(not(any(test, feature = "proptest-impl")))]
                        false,
```

Reply:
Nope.

Reply:
Yea, sorry about that. I will try something else.